editoast/editoast_derive/src/annotate_units.rs

```diff
@@ -0,0 +1,82 @@
+use darling::Result;
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::{parse_quote, DeriveInput, LitStr};
+
+const OPTIONAL_SUFFIX: &str = "option";
+struct UnitType {
+    unit: String,
+    optional: bool,
+}
+
+impl UnitType {
+    fn new(path: LitStr) -> Self {
+        let path = path.value();
+        let paths: Vec<_> = path.split("::").collect();
+        if paths.len() >= 2 && paths.last() == Some(&OPTIONAL_SUFFIX) {
+            Self {
+                unit: paths[paths.len() - 2].to_owned(),
+                optional: true,
+            }
+        } else {
+            Self {
+                unit: paths
+                    .last()
+                    .expect("String::split to return at least an empty string")
+                    .to_string(),
+                optional: false,
+            }
+        }
+    }
```

Comment:
A suggestion to use functions of `str` to make the code more concise (see [Rust playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=55a1f62250405a705c4ba2832d369372)) by using `strip_suffix` and `rsplit`.